### PR TITLE
feat:CO-91: handle updated BE response

### PR DESCRIPTION
**Pull request by arjun-at-roam**
## PR Area

- [x] Frontend

## Type of change

_What kind of change does this PR introduce?_

- [x] Feature

## Description

* remove `hybrid` product type as it no longer exists
* add `payment-holiday` product type
* remove all properties that are not shown to user from `ResponseLanguage.ts`
* new function `getProductValue`
  * prefix `$` if required
  * suffix `months` if required
  * decide between`months`/`month`
  * convert string whole numbers to stings with 2 decimal places

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots

image

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/lang/ResponseLanguage.ts [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
+    label: 'Indicative minimum monthly repayments*',
+    unit: '$',
   },
   {
-    name: 'remainderInterestPeriodMontlyRepayment',
-    label: 'Reminder interest period monthly repayment',
+    name: 'indicativeMonthly',
+    label: 'Indicative repayment to pay before Interest free period expires ^',
+    unit: '$',
   },
```
Comment (arjun-at-roam) [inline src/lang/ResponseLanguage.ts:15]: these are the only properties that will be shown to user.

File: src/utils/utils.ts [hunk 2]
```diff
@@ -162,6 +162,22 @@ export const getProductLabel = (nameKey: string): string => {
   return nameLabelPair ? nameLabelPair.label : nameKey
 }
 
+export const getProductValue = (nameKey: string, value: string): string => {
+  const nameLabelPair = ProductLanguage.find(item => item.name === nameKey)
+  if (nameLabelPair)
+    switch (nameLabelPair.unit.toLowerCase()) {
+      case '$':
+        return `${nameLabelPair.unit}${parseFloat(value).toFixed(2)}`
```
Comment (arjun-at-roam) [inline src/utils/utils.ts:170]: `50` -> `$50.00`
or
`0.01` -> `$0.01`

File: src/utils/utils.ts [hunk 3]
```diff
@@ -162,6 +162,22 @@ export const getProductLabel = (nameKey: string): string => {
...
+      case 'months': {
+        const unit = `month${value === '1' ? '' : 's'}`
+        return `${value} ${unit}`
```
Comment (arjun-at-roam) [inline src/utils/utils.ts:173]: `8` -> `8 months`
or
`1` -> `1 month`

File: src/lang/ResponseLanguage.ts [hunk 4]
```diff
@@ -1,22 +1,14 @@
 export const ProductLanguage = [
```
Comment (antoninroam) [inline src/lang/ResponseLanguage.ts]: would be good to add type or interface for this const

**Review by antoninroam — APPROVED** [review]

**Review by jpyepez-roam — APPROVED** [review]
LGTM!

File: src/lang/ResponseLanguage.ts (hunk 4 above)
Comment (ghost) [reply src/lang/ResponseLanguage.ts]: I agree with @antoninroam. Also, nit, this should be camelCased not PascalCased

**Review by ghost — APPROVED** [review]

File: src/lang/ResponseLanguage.ts (hunk 4 above)
Comment (arjun-at-roam) [reply src/lang/ResponseLanguage.ts]: nice catch. Thank you 👍

**Pull request merged**